packages/app/src/specs/generators/ExpandableFileChooser.cy.tsx

```diff
@@ -205,14 +205,14 @@ describe('<ExpandableFileChooser />', () => {
     .type(newExtension, { delay: 0 })
 
     // Validate it's in there
-    .get(extensionInputSelector)
+    cy.get(extensionInputSelector)
     .should('have.value', newExtension)
 
     // debounce should cause this to hit
-    .get('@onUpdateExtensionSpy').should('not.have.been.calledWith', newExtension)
+    cy.get('@onUpdateExtensionSpy').should('not.have.been.calledWith', newExtension)
```

Comment:
Since `.should()` is now a query, continuing to chain additional queries off of it (like .get()) would continue to re-execute the whole set of commands.

Giant chains of commands without a `cy.` anywhere are not great practice, now that queries make Cypress' retry logic more powerful.

Reply:
They're required, because the chain is now entirely composed of queries. You have a chain that looks like:

`[<input>, .get('@spy'), .should('not.have.been.called'), get('@spy'), .should('have.been.called')]`

which can never pass, because the spy can't even have been called and not have been called at the same time. There's no non-query commands to break it up anymore.

It's not clearly called out in the migration guide because it's not a `.should()` specific problem, it's related to how queries work in general. We ran into the same changes as part of 12.x, and we'll run into them again if we ever migrate additional queries to be non-queries.

With that said, it is fairly non-obvious, and we should figure out a way to tell users about it more clearly than my above explanation. It's fairly similar to https://docs.cypress.io/guides/references/migration-guide#Behavior-Changes-in-Alias-Resolution, so a new section in the same vein would probably be helpful.

Reply:
I've opened https://github.com/cypress-io/cypress/issues/25813 to think about and discuss some related work (it may be possible to avoid requiring users to manually do this breaking up).

Reply:
I've also updated the migration guide with a new section:

> #### Chains that include assertions now retry "all the way through"
> Previously, assertions were non-query commands, meaning that they served as
> "breakpoints" in a chain of queries, and queries before them did not re-run.

plus some more examples and explanations. See docs PR for details.

Reply:
Its still not clear to me why
```
cy.get(extensionInputSelector)
    .should('have.value', newExtension)
```
 needed to be broken form the change from `.get` to `cy.get()`. I would have expected `.type()` to have broken the chain and these commands to retry as expected without the break.